azure/services/aso/service_test.go

```diff
@@ -286,7 +286,7 @@ func TestServiceReconcile(t *testing.T) {
 			}, nil
 		}
 		reconciler.EXPECT().DeleteResource(gomockinternal.AContext(), deleteMe, serviceName).Return(deleteErr)
-		reconciler.EXPECT().DeleteResource(gomockinternal.AContext(), deleteMeToo, serviceName).Return(deleteErr)
+		reconciler.EXPECT().DeleteResource(gomockinternal.AContext(), deleteMeToo, serviceName).Return(azure.NewOperationNotDoneError(&infrav1.Future{}))
```

Comment:
This change isn't related to the rest of this particular commit. I just noticed this would better ensure that the error precedence is working as expected.